### PR TITLE
Cn 886 support docker image artifact

### PR DESCRIPTION
**Pull request by billiford**
This PR:
- Adds support for the `docker/image` artifact type

### CODE REVIEW

File: pkg/kubernetes/artifact.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	)
+
+	if len(artifacts) == 0 {
+		return nil
+	}
+
+	if strings.EqualFold(u.GetKind(), "pod") {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "containers")
+	} else {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "containers")
```
Comment (dmrogers7) [inline pkg/kubernetes/artifact.go:32]: I failed to give full requirements in the story, but will need to get `initContainers` as well

See the [clouddriver code](https://github.com/spinnaker/clouddriver/blob/58ab154b0ec0d62772201b5b319af349498a4e3f/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java#L166)

File: pkg/kubernetes/artifact_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+			It("updates the container image to contain the reference", func() {
+				Expect(err).To(BeNil())
+				p := NewPod(resource.Object)
+				containers := p.Object().Spec.Containers
+				Expect(containers[0].Image).To(Equal("gcr.io/test-project/test-container-image:v1.0.0"))
+			})
+		})
+
+		When("the kind is deployment", func() {
+			BeforeEach(func() {
```
Comment (dmrogers7) [inline pkg/kubernetes/artifact_test.go:124]: Can you write a `when a container image does not match artifact image` test?  Define the resource with two containers/images and validate that only the one image is replaced

```go
		resource = &unstructured.Unstructured{
			Object: map[string]interface{}{
				"kind":       "Deployment",
				"apiVersion": "apps/v1",
				"spec": map[string]interface{}{
					"template": map[string]interface{}{
						"spec": map[string]interface{}{
							"containers": []interface{}{
								map[string]interface{}{
									"name":  "test-container-name",
									"image": "gcr.io/test-project/test-container-image",
								},
								map[string]interface{}{
									"name":  "another-test-container-name",
									"image": "gcr.io/test-project/another-test-container-image",
								},
							},
						},
					},
				},
			},
		}
	})		
```

**Review by dmrogers7 — CHANGES_REQUESTED** [review]

File: pkg/kubernetes/artifact.go (hunk 1 above)
Comment (billiford) [reply pkg/kubernetes/artifact.go:32]: Arg! I missed that.

File: pkg/kubernetes/artifact_test.go (hunk 2 above)
Comment (billiford) [reply pkg/kubernetes/artifact_test.go:124]: Sure thing.

**Comment by billiford** [comment]
- changed function name to `BindDockerImageArtifacts`
- added support for init containers
- added full test section for init containers
- added new tests for multiple containers/initContainers

**Review by dmrogers7 — APPROVED** [review]
LGTM!

**Pull request merged**